### ISSUE TITLE
Shard Respawn Max Tries

### ISSUE THREAD

**Issue by cloewen8** [discordjs/discord.js#2882]
**Is your feature request related to a problem? Please describe.**
When a shard is unable to start, and respawn is true, the shard continues to respawn shards indefinitely until Discord resets the token.

**Describe the ideal solution**
Ideally, I would like to be able to,
1. Stop shards from respawning if they continuously die.
2. Proceed with an alternative action if shards are unable to respawn.

One possible solution:
- A maximum number of retries and a minimum alive time is provided as options.
- The ShardingManager tracks tries.
- If a shard fails to stay alive for a minimum number of milliseconds, the try is recorded.
- If the number of tries exceeds the retries, stop respawning shards and emit an event on the ShardingManager.
- If a shard stays alive for the minimum number of milliseconds, reset the tries.

**Describe alternatives you've considered**
- Setting respawn to false and wrapping the shards in protected blocks (such as try-catch) to filter what is thrown. - Assumes all errors can be trapped. What should be filtered would depend on the implementation.
- Setting respawn to false in development environments. - Assumes everyone on our team is properly testing their changes before deploying to a production environment.

**Additional context**
I would be willing to work on a pull request for this if a solution can be agreed on.

**Comment by cloewen8** [comment]
I don't see how this is a bug. I am requesting a feature to gain more fine grained control. Was this feature already implemented? Or does it relate to a bug I'm not noticing?

**Comment by Gawdl3y** [comment]
You already can control it - disable auto-respawn and listen for the relevant events yourself.

**Comment by Poly-Pixel** [comment]
There's been no activity on this issue for over 2 years, and it seems to be resolved - I think this issue should be closed

**Comment by cloewen8** [comment]
I don't believe the last suggestion resolved the issue if that is what you are referring to. I have not used discord.js in a while, perhaps I am missing a commit.

**Comment by Poly-Pixel** [comment]
> You already can control it - disable auto-respawn and listen for the relevant events yourself.

I might be understanding this wrong, but this method allows you to do what you described.

**Comment by cloewen8** [comment]
> 
> 
> > You already can control it - disable auto-respawn and listen for the relevant events yourself.
> 
> I might be understanding this wrong, but this method allows you to do what you described.

I'd like to direct your attention to the previous comment:

> Thanks for the idea, didn't think of doing it manually. If I do implement this manually, I'll post a gist to it. I still feel like this is a feature a broad range of bot owners would find useful.

It is a solution, but it needs to be implemented manually. I don't feel like that is sufficient to consider the issue resolved. If the maintainers disagree, they are free to close this issue. I'm not using discord.js, I don't mind.

As an example, it is like requesting Discord add support for commands, despite bots already having commands. Having commands is great! But officially supported commands are better.
